File: unit-3/sprint-2/evaluation/app/src/main/java/com/example/evaluation/Adaptor.java
package com.example.evaluation;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class Adaptor extends RecyclerView.Adapter<Holder> {

    private List<ResponseModel> responseModelList;

    public Adaptor(List<ResponseModel> responseModelList) {
        this.responseModelList = responseModelList;
    }

    @NonNull
    @Override
    public Holder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_layout2, parent, false);
        return new Holder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull Holder holder, int position) {
        ResponseModel responseModel = responseModelList.get(position);
        holder.setData(responseModel);
    }


    @Override
    public int getItemCount() {
        return responseModelList.size();
    }
}

File: unit-3/sprint-4/evaluation/Music Preview/app/src/main/java/com/example/musicpreview/holderandadaptor/Adaptor.kt
package com.example.musicpreview.holderandadaptor

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.musicpreview.ApiService.ResponseDTO
import com.example.musicpreview.R

class Adaptor(private val responseDTO: List<ResponseDTO>) : RecyclerView.Adapter<Holder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Holder {
   val view = LayoutInflater.from(parent.context).inflate(R.layout.item_layout,parent,false);
        return Holder(view);
    }

    override fun onBindViewHolder(holder: Holder, position: Int) {

    }

    override fun getItemCount(): Int {
     return responseDTO.size;
    }
}